tests/MCPClient/test_policy_check.py

```diff
@@ -0,0 +1,592 @@
+    preservation_file_format_version: models.FileFormatVersion,
+    format: fprmodels.Format,
+    format_version: fprmodels.FormatVersion,
+    shared_directory_path: str,
+) -> None:
+    job = mock.Mock(
+        args=[
+            "policy_check",
+            preservation_file.currentlocation.decode(),
+            str(preservation_file.uuid),
+            str(sip.uuid),
+            str(shared_directory_path),
+            preservation_file.filegrpuse,
+        ],
+        JobContext=mock.MagicMock(),
+        spec=Job,
+    )
+
+    policy_check.call([job])
+
+    job.set_status.assert_called_once_with(0)
+    assert job.pyprint.mock_calls == [
+        mock.call(
+            f"File {preservation_file.uuid} is not a preservation derivative; not performing a policy check."
+        )
+    ]
+
+
+@pytest.mark.django_db
+@mock.patch("policy_check.executeOrRun")
+def test_policy_check_verifies_file_type_is_access(
+    execute_or_run: mock.Mock,
+    derivation_for_access: models.Derivation,
+    access_file: models.File,
+    sip: models.SIP,
+    fprule_policy_check: fprmodels.FPRule,
+    access_file_format_version: models.FileFormatVersion,
+    format: fprmodels.Format,
+    format_version: fprmodels.FormatVersion,
+    shared_directory_path: str,
+) -> None:
+    job = mock.Mock(
+        args=[
+            "policy_check",
+            access_file.currentlocation.decode(),
+            str(access_file.uuid),
+            str(sip.uuid),
+            str(shared_directory_path),
+            "access",
+        ],
+        JobContext=mock.MagicMock(),
+        spec=Job,
+    )
+    expected_stdout = json.dumps(
+        {
+            "eventOutcomeInformation": "pass",
+            "eventOutcomeDetailNote": "a note",
+        }
+    )
+    execute_or_run.return_value = (0, expected_stdout, "")
+
+    policy_check.call([job])
+
+    job.set_status.assert_called_once_with(0)
+    assert job.pyprint.mock_calls == [
+        mock.call("Running", ""),
+        mock.call(
+            f"Command {fprule_policy_check.command.description} completed with output {expected_stdout}"
+        ),
+        mock.call(
+            f"Creating policy checking event for {access_file.currentlocation.decode()} ({access_file.uuid})"
+        ),
+    ]
+
+
+@pytest.mark.django_db
+@mock.patch("policy_check.executeOrRun")
+def test_policy_check_fails_if_file_is_not_access_derivative(
+    execute_or_run: mock.Mock,
+    access_file: models.File,
+    sip: models.SIP,
+    fprule_policy_check: fprmodels.FPRule,
+    access_file_format_version: models.FileFormatVersion,
+    format: fprmodels.Format,
+    format_version: fprmodels.FormatVersion,
+    shared_directory_path: str,
+) -> None:
+    job = mock.Mock(
+        args=[
+            "policy_check",
+            access_file.currentlocation.decode(),
+            str(access_file.uuid),
+            str(sip.uuid),
+            str(shared_directory_path),
+            "access",
```

Comment:
The preservation case of this test is using the `filegrpuse` attribute instead of a hard coded string.